crates/wallet/tests/wallet.rs

```diff
@@ -4239,3 +4239,52 @@ fn test_tx_builder_is_send_safe() {
     let (mut wallet, _txid) = get_funded_wallet_wpkh();
     let _box: Box<dyn Send + Sync> = Box::new(wallet.build_tx());
 }
+
+#[test]
+fn test_wallet_transactions_relevant() {
+    let (mut test_wallet, _txid) = get_funded_wallet_wpkh();
+    let relevant_tx_count_before = test_wallet.transactions().count();
+    let full_tx_count_before = test_wallet.tx_graph().full_txs().count();
+    let chain_tip = test_wallet.local_chain().tip().block_id();
+    let canonical_tx_count_before = test_wallet
+        .tx_graph()
+        .list_canonical_txs(test_wallet.local_chain(), chain_tip)
+        .count();
+
+    // add not relevant transaction to test wallet
+    let (other_external_desc, other_internal_desc) = get_test_tr_single_sig_xprv_and_change_desc();
+    let (other_wallet, other_txid) = get_funded_wallet(other_internal_desc, other_external_desc);
+    let other_tx_node = other_wallet.get_tx(other_txid).unwrap().tx_node;
+    let other_tx_confirmationblocktime = other_tx_node.anchors.iter().last().unwrap();
+    let other_tx_update = TxUpdate {
+        txs: vec![other_tx_node.tx],
+        txouts: Default::default(),
+        anchors: [(*other_tx_confirmationblocktime, other_txid)].into(),
+        seen_ats: [(other_txid, other_tx_confirmationblocktime.confirmation_time)].into(),
+    };
+    let test_wallet_update = Update {
+        last_active_indices: Default::default(),
+        tx_update: other_tx_update,
+        chain: None,
+    };
```

Comment:
This can be simplified by getting a `TxUpdate` directly from the other tx graph.

```diff
-    let other_tx_node = other_wallet.get_tx(other_txid).unwrap().tx_node;
-    let other_tx_confirmationblocktime = other_tx_node.anchors.iter().last().unwrap();
-    let other_tx_update = TxUpdate {
-        txs: vec![other_tx_node.tx],
-        txouts: Default::default(),
-        anchors: [(*other_tx_confirmationblocktime, other_txid)].into(),
-        seen_ats: [(other_txid, other_tx_confirmationblocktime.confirmation_time)].into(),
-    };
     let test_wallet_update = Update {
-        last_active_indices: Default::default(),
-        tx_update: other_tx_update,
-        chain: None,
+        tx_update: other_wallet.tx_graph().clone().into(),
+        ..Default::default()
     };
```

Reply:
Good suggestion, fixed in #1785.